code/game/machinery/Sleeper.dm

```diff
@@ -449,25 +453,23 @@
 /obj/machinery/sleeper/ex_act(severity)
 	if(filtering)
 		toggle_filter()
+	var/mob/M = src.occupant
 	switch(severity)
 		if(1.0)
-			for(var/atom/movable/A as mob|obj in src)
-				A.forceMove(src.loc)
-				ex_act(severity)
+			go_out()
+			M.ex_act(severity)
 			qdel(src)
 			return
 		if(2.0)
 			if(prob(50))
-				for(var/atom/movable/A as mob|obj in src)
-					A.forceMove(src.loc)
-					ex_act(severity)
+				go_out()
+				M.ex_act(severity)
 				qdel(src)
 				return
 		if(3.0)
 			if(prob(25))
-				for(var/atom/movable/A as mob|obj in src)
-					A.forceMove(src.loc)
-					ex_act(severity)

```

Comment:
To clarify, I was refering to the original version, it called ex_act() in a for loop, which caused a recursion chain until the sleeper was empty and that by removing the loop, nothing would terminate the recursion and thus there would be an infinite loop if ex_act() was called

Anyway, after rereading his suggestion, I see what he actually was referring to with his comment and considering that the contents could potentially have something that should chain explode, it will probably work better that way.